test/MorseL.Sockets.Tests/MorseLHttpMiddlewareTests.cs

```diff
@@ -0,0 +1,165 @@
+        }
+
+        [Fact]
+        public async Task WebSocketRequest_DoesNotInvokeNextRequestDelegate()
+        {
+            var sut = CreateMorseLHttpMiddleware();
+
+            await sut.Invoke(Mocker.HttpContextMock.Object);
+
+            RequestDelegateMock.Verify(m => m.Next(It.IsAny<HttpContext>()), Times.Never);
+        }
+
+        [Fact]
+        public async Task WebSocketRequest_ShouldBeAddedToConnectionManager()
+        {
+            var sut = CreateMorseLHttpMiddleware();
+
+            await sut.Invoke(Mocker.HttpContextMock.Object);
+
+            Mocker.WebSocketConnectionManagerMock.Verify(m => m.AddConnection(It.IsAny<IChannel>()), Times.Once);
+        }
+
+        [Fact]
+        public async Task WebSocketRequest_ShouldBeRemovedFromConnectionManager_IfSocketIsNotOpen()
+        {
+            Mocker.WebSocketMock.Setup(m => m.State).Returns(WebSocketState.None);
+
+            var sut = CreateMorseLHttpMiddleware();
+
+            await sut.Invoke(Mocker.HttpContextMock.Object);
+
+            Mocker.WebSocketConnectionManagerMock
+                .Verify(m => m.RemoveConnection(It.Is<string>(s => s == Mocker.ConnectionId)), Times.Once);
+        }
+
+        [Fact]
+        public async Task WebSocketRequest_ShouldBeRemovedFromConnectionManager_IfCancellationIsRequested()
+        {
+            var cts = new CancellationTokenSource();
+            cts.Cancel();
+
+            var sut = CreateMorseLHttpMiddleware(cts);
+
+            await sut.Invoke(Mocker.HttpContextMock.Object);
+
+            Mocker.WebSocketConnectionManagerMock
+                .Verify(m => m.RemoveConnection(It.Is<string>(s => s == Mocker.ConnectionId)), Times.Once);
+        }
+
+        [Fact]
+        public async Task WebSocketRequest_ShouldBeRemovedFromConnectionManager_IfExceptionThrownInReceiveLoop()
+        {
+            Mocker.WebSocketMock.Setup(m => m.State).Returns(WebSocketState.Open);
+            Mocker.WebSocketMock.Setup(m => m.ReceiveAsync(It.IsAny<ArraySegment<byte>>(), It.IsAny<CancellationToken>()))
+                .Throws<Exception>();
+
+            var sut = CreateMorseLHttpMiddleware();
+
+            await sut.Invoke(Mocker.HttpContextMock.Object);
+
+            Mocker.WebSocketConnectionManagerMock
+                .Verify(m => m.RemoveConnection(It.Is<string>(s => s == Mocker.ConnectionId)), Times.Once);
+        }
+
+        [Fact]
+        public async Task WebSocketRequest_ShouldBeRemovedFromConnectionManager_IfCloseMessageIsReceived()
+        {
+            Mocker.WebSocketMock.Setup(m => m.State).Returns(WebSocketState.Open);
+
+            var sut = CreateMorseLHttpMiddleware();
+
+            // Fire and forget since this is essentially just an infinite loop
+#pragma warning disable CS4014 // Because this call is not awaited, execution of the current method continues before the call is completed
+            sut.Invoke(Mocker.HttpContextMock.Object);
```

Comment:
You can just avoid the warning this way:
```suggestion
            var _ = sut.Invoke(Mocker.HttpContextMock.Object);
```

Reply:
Ah. That's definitely nicer. The warning disable is maybe more explicit? I'm definitely open to both.